packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -103,6 +103,14 @@ enum HostArtifact {
   pubExecutable,
 }
 
+// TODO(knopp): Remove once darwin artifacts are moved out of darwin-x64
```

Comment:
I don't think we're going to move darwin artifacts out of darwin-x64. We'll still have different engine builds for darwin-x64 and darwin-arm - the target platform is best though of as a convenience for how the engine builds think of targets that was incorrectly extended to how our users interact with the platforms.

Reply:
Why do you need different engine builds? Can you just have universal FlutterMacOS.framework (just like you have universal Flutter.framework for ios)? The PR already implements framework thinning. 

I assumed that all the artifacts would be distributed as universal in the end.

Reply:
No better answer than "that is how the engine builds are configured currently".

Reply:
Well, sure, but I assumed that at some point there's going to be work done to build the engine for darwin-x64, darwin-arm64 and lipo them together. Isn't that the whole purpose of not having separate `TargetPlatform.darwin_x64` and `.darwin_arm64`?

Somehow this is done for iOS, why can't it be done for macOS? (plus flutter_tester, font-subset and gen_snapshot).

`CachedArtifacts` doesn't know about `DarwinArchs` so I don't quite see how it would choose the right folder if you want to distribute darwin-arm64 build separately.

Reply:
You could certainly do that work, but there is no point in changing the cache structure in the tool before that is done

Reply:
I didn't assume this would happen before having universal artifacts. There's no point in doing that. I'll clarify that in the TODO comment.

Reply:
Quasi related for distribution but I suggested combining the macOS framework into the iOS Flutter.xcframework https://github.com/flutter/flutter/issues/70413 though considering how unfortunately different the two platform projects look, and how manual it was to get the right framework slice out of the iOS xcframework, I'm not now sure if that will be very valuable.  In any case, we should distribute a universal macOS framework.  But no one has worked on that yet.

Reply:
Well, I have no clue how flutter builds work on server side so I'm not sure I can help with that :) However with recent changes landed in buildroot and engine it is now possible to cross compile flutter for darwin-arm64. But it requires manually editing `build.gn`. I think the next step would be to add darwin cross compilation to `flutter/tools/gn`. I'll open an issue for it.